### PR TITLE
Update creature_template and creature_classlevelstats

### PR DESCRIPTION
**Pull request by Hacki95**
## Description
-Updated creature_template and creature_classlevelstats to fit the changes made by https://github.com/azerothcore/azerothcore-wotlk/pull/4749
-Removed mindamage, maxdamage, minrangedamage, maxrangedamage, attackpower, rangedattackpower from creature_template.
-Added a description for BaseVariance and RangeVariance
-Added the damage calculation formulas.
-Updated base_damage, damage_exp1 and damage_exp2 in creature_classlevelstats

## Related Issue
https://github.com/azerothcore/azerothcore-wotlk/pull/4749
https://github.com/azerothcore/wiki/issues/407

## Motivation and Context
-Update the wiki and remove some wrong information.

## How Has This Been Tested?
-Checked it in my local branch, looks alright although I am not a formating expert. If someone has any suggestions to make it better I will gladly update this PR.

### CODE REVIEW

File: docs/creature_template.md [hunk 1]
```diff
@@ -27,12 +27,11 @@ This table contains the description of creatures. Each spawned creature is an in
...
-| [mindmg](#mindmg)                               | float                 | NO   |     | 0       |       |                                      |
-| [maxdmg](#maxdmg)                               | float                 | NO   |     | 0       |       |                                      |
 | [dmgschool](#dmgschool)                         | tinyint(4)            | NO   |     | 0       |       |                                      |
-| [attackpower](#attackpower)                     | int(10) unsigned      | NO   |     | 0       |       |                                      |
 | [BaseAttackTime](#baseattacktime)               | int(10) unsigned      | NO   |     | 0       |       |                                      |
 | [RangeAttackTime](#rangeattacktime)             | int(10) unsigned      | NO   |     | 0       |       |                                      |
+| [BaseVariance](#BaseVariance)                   | float                 | NO   |     | 1       |       |                                      |
+| [RangeVariance](#RangeVariance)                 | float                 | NO   |     | 1       |       |                                      |
```
Comment (Kitzunu) [inline docs/creature_template.md:34]: use only small letters within () otherwise link will be broken

File: docs/creature_template.md [hunk 2]
```diff
@@ -273,6 +255,18 @@ This is the base time that determines how long a creature must wait between mele
...
 
+#### BaseVariance
+
+Value to customize the creature's damage output. See [DamageModifier](http://www.azerothcore.org/wiki/creature_template#DamageModifier)
+
+Non-custom creatures should always leave this at 1.
+
+#### RangeVariance
+
+Value to customize the creature's damage output. See [DamageModifier](http://www.azerothcore.org/wiki/creature_template#DamageModifier)
```
Comment (Kitzunu) [inline docs/creature_template.md:266]: #damagemodifier

File: docs/creature_template.md [hunk 3]
```diff
@@ -577,7 +557,21 @@ Used to modify the base Level/Class armor of a creature.
...
+
+damage_base comes from the creature_classless table and takes its value either from [damage_base](https://www.azerothcore.org/wiki/creature_classlevelstats#damage_base), [damage_exp1](https://www.azerothcore.org/wiki/creature_classlevelstats#damage_exp1) or [damage_exp2](https://www.azerothcore.org/wiki/creature_classlevelstats#damage_exp2) according to the creature's value in [exp](http://www.azerothcore.org/wiki/creature_template#exp) (0 = base_damage, 1 = damage_exp1, 2 = damage_exp2).
+
+BaseAttackTime is either [BaseAttackTime](https://www.azerothcore.org/wiki/creature_template#baseattacktime) or [RangeAttackTime](https://www.azerothcore.org/wiki/creature_template#rangeattacktime) depending on the type of attack.
+
+attackpower is either [attackpower](https://www.azerothcore.org/wiki/creature_classlevelstats#attackpower) or [rangedattackpower](https://www.azerothcore.org/wiki/creature_classlevelstats#rangedattackpower) depending on the type of attack.
+
+BaseVariance is either [BaseVariance](https://www.azerothcore.org/wiki/creature_template#BaseVariance) or [RangeVariance](https://www.azerothcore.org/wiki/creature_template#RangeVariance) depending on the type of attack.
```
Comment (Kitzunu) [inline docs/creature_template.md:573]: Use only the small links, so for example [BaseVariance](#basevariance)

File: docs/creature_template.md [hunk 4]
```diff
@@ -577,7 +557,21 @@ Used to modify the base Level/Class armor of a creature.
 
 #### DamageModifier
 
-TODO!
+Used to modify the Minimum/Maximum damage of a creature.
```
Comment (Kitzunu) [inline docs/creature_template.md:560]: useless tbh

Comment (Kitzunu) [inline docs/creature_template.md:560]: useless comment tbh.
it doesn't matter what it did in the past, what matters is what it does now

Comment (Hacki95) [reply docs/creature_template.md:560]: Maybe it's more of a grammar mistake. I mean it like "(This value is) used to...". Will try to make it a bit easier to understand.

File: docs/creature_template.md (hunk 3 above)
Comment (Hacki95) [reply docs/creature_template.md:573]: Alright will add those later.

File: docs/creature_template.md (hunk 2 above)
Comment (Hacki95) [reply docs/creature_template.md:266]: Will edit this.

File: docs/creature_template.md (hunk 4 above)
Comment (Kitzunu) [reply docs/creature_template.md:560]: "This field is ued to..." could work

Comment (Hacki95) [reply docs/creature_template.md:560]: Should I change it for the other Modifiers too? Thats where I got the sentence from.

Comment (Kitzunu) [reply docs/creature_template.md:560]: Actually you can leave it as-is

**Comment by FrancescoBorzi** [comment]
@Kitzunu @Hacki95 is this ready to be merged?

**Comment by Hacki95** [comment]
@FrancescoBorzi I think I am done. I shortened all the other links in the document too while I was at it. @Kitzunu could you recheck if I missed anything?

**Comment by Hacki95** [comment]
Also if I want to reference another document is (otherdocument#field) the right way?

File: docs/creature_template.md [hunk 5]
```diff
@@ -27,12 +27,11 @@ This table contains the description of creatures. Each spawned creature is an in
...
-| [mindmg](#mindmg)                               | float                 | NO   |     | 0       |       |                                      |
-| [maxdmg](#maxdmg)                               | float                 | NO   |     | 0       |       |                                      |
 | [dmgschool](#dmgschool)                         | tinyint(4)            | NO   |     | 0       |       |                                      |
-| [attackpower](#attackpower)                     | int(10) unsigned      | NO   |     | 0       |       |                                      |
 | [BaseAttackTime](#baseattacktime)               | int(10) unsigned      | NO   |     | 0       |       |                                      |
 | [RangeAttackTime](#rangeattacktime)             | int(10) unsigned      | NO   |     | 0       |       |                                      |
+| [BaseVariance](#baseVariance)                   | float                 | NO   |     | 1       |       |                                      |
+| [RangeVariance](#rangeVariance)                 | float                 | NO   |     | 1       |       |                                      |
```
Comment (Kitzunu) [inline docs/creature_template.md:34]: basevariance, rangevariance in links

File: docs/creature_template.md [hunk 6]
```diff
@@ -273,9 +255,21 @@ This is the base time that determines how long a creature must wait between mele
...
+#### RangeVariance
+
+Value to customize the creature's damage output. See [DamageModifier](#damageModifier).
+
+Non-custom creatures should always leave this at 1.
+
 #### unit_class
 
-This is the creature's class, and it dictates levels of health and mana. Also note that health and mana will change according to [exp](http://www.azerothcore.org/wiki/creature_template#exp), [HealthModifier](http://www.azerothcore.org/wiki/creature_template#HealthModifier), and [ManaModifier](http://www.azerothcore.org/wiki/creature_template#ManaModifier). Not setting this value will report a minor warning in the "DB_Errors.log".
+This is the creature's class, and it dictates levels of health and mana. Also note that health and mana will change according to [exp](#exp), [HealthModifier](#healthModifier), and [ManaModifier](#manaModifier). Not setting this value will report a minor warning in the "DB_Errors.log".
```
Comment (Kitzunu) [inline docs/creature_template.md:272]: all small within ()

File: docs/creature_template.md [hunk 7]
```diff
@@ -487,15 +467,15 @@ This field can control whether a mob is minable or herbable or lootable by engin
 
 #### lootid
 
-The ID of the loot template ID that this creature should use to generate loots. See [creature_loot_template.entry](http://www.azerothcore.org/wiki/loot_template#loot_template-Entry)
+The ID of the loot template ID that this creature should use to generate loots. See [creature_loot_template.entry](loot_template#loot_template-Entry)
```
Comment (Kitzunu) [inline docs/creature_template.md:470]: same as above

File: docs/creature_template.md [hunk 8]
```diff
@@ -487,15 +467,15 @@ This field can control whether a mob is minable or herbable or lootable by engin
...
 
 #### pickpocketloot
 
-The ID of the pickpocketing loot template that this creature should use to generate pickpocketing loots. See [pickpocketing_loot_template.entry](http://www.azerothcore.org/wiki/loot_template#loot_template-Entry)
+The ID of the pickpocketing loot template that this creature should use to generate pickpocketing loots. See [pickpocketing_loot_template.entry](loot_template#loot_template-Entry)
```
Comment (Kitzunu) [inline docs/creature_template.md:474]: same as above

File: docs/creature_template.md [hunk 9]
```diff
@@ -487,15 +467,15 @@ This field can control whether a mob is minable or herbable or lootable by engin
...
 
 #### skinloot
 
-The ID of the skinning loot template that this creature should use to generate skinning loots. See [skinning_loot_template.entry](http://www.azerothcore.org/wiki/loot_template#loot_template-Entry)
+The ID of the skinning loot template that this creature should use to generate skinning loots. See [skinning_loot_template.entry](loot_template#loot_template-Entry)
```
Comment (Kitzunu) [inline docs/creature_template.md:478]: same as above

File: docs/creature_template.md [hunk 10]
```diff
@@ -557,7 +537,7 @@ Controls where the creature can move and attack.
 
 (This is a bitmask. You can add values together: 1+4=5 would make the creature walk on ground and fly.)
 
-**Note:** If your vehicle is a flying vehicle then your accessory **MUST** have it's [InhabitType](http://www.azerothcore.org/wiki/creature_template#InhabitType) set to (4 - Flying). This being if you set it for both ground and flying it will spawn on the ground if the vehicle is initially spawned on the ground.
+**Note:** If your vehicle is a flying vehicle then your accessory **MUST** have it's [InhabitType](#inhabitType) set to (4 - Flying). This being if you set it for both ground and flying it will spawn on the ground if the vehicle is initially spawned on the ground.
```
Comment (Kitzunu) [inline docs/creature_template.md:540]: same as above

**Comment by Kitzunu** [comment]
This is the last, I promise :D

**Comment by Hacki95** [comment]
> 
> 
> This is the last, I promise :D

All good! I misunderstood the links, I thought the just needed to start with a lowercase. 

I think I got them all now

**Review by Kitzunu — APPROVED** [review]

**Comment by Kitzunu** [comment]
=)

**Pull request merged**